### PR TITLE
Fix maps domain brittleness

### PR DESCRIPTION
**Pull request by kubark42**
The current Google API approach is very brittle to the crypto key and GCS will crash if the crypto key is wrong. This gets the base domain name from the organization name in a more robust manner, so `www.taulabs.org` returns the same results as `forums.taulabs.org` as `taulabs.org`, etc...

In the event that the key is just plain wrong, then the decrypt function will thrown an error, which is caught in the parent. The parent will send a debug message instead of GCS segfaulting.

It could be arguable, however, that we should do a `qFatal()` instead of `qDebug()`, as not having correct provider strings could lead to system instability elsewhere. Still, in limited testing on OSX 10.10 having the function return early did _not_ lead to crashes, or even any immediate decrease in functionality.

Fixes problem seen in https://github.com/TauLabs/TauLabs/pull/2106#issuecomment-163786632. Once this is in we can merge https://github.com/TauLabs/TauLabs/pull/2106.

### CODE REVIEW

File: ground/gcs/src/libs/tlmapcontrol/core/providerstrings.cpp [hunk 1]
```diff
@@ -67,13 +68,31 @@ QString ProviderStrings::decrypt(QString str)
         ++pos;
     }
 
-    return QString::fromUtf8(qUncompress(ba));
+    QString decryptedString = QString::fromUtf8(qUncompress(ba));
+    if (decryptedString.isEmpty()) {
+        throw std::domain_error(QString("Decrypted to an empty vector!. Input vector: ").append(inputString).toStdString());
```
Comment (peabody124) [inline ground/gcs/src/libs/tlmapcontrol/core/providerstrings.cpp:73]: What happens when this is thrown? What catches it and is it shown to the user somehow?

Comment (peabody124) [inline ground/gcs/src/libs/tlmapcontrol/core/providerstrings.cpp:73]: Nevermind. See it later ;)

**Comment by peabody124** [comment]
Code reviews well. About to test.

File: ground/gcs/src/libs/tlmapcontrol/core/providerstrings.cpp (hunk 1 above)
Comment (kubark42) [inline ground/gcs/src/libs/tlmapcontrol/core/providerstrings.cpp:73]: Is there a standard way to make this clear in comments? It seems unfortunate that you had to hunt through the code to find out what the catch was going to be. It sort of is like a `goto` in the sense that it can make it hard to follow the program's progression.
